### PR TITLE
Roll 51 build exchange card path

### PR DESCRIPTION
**Pull request by brandlxee11**

### CODE REVIEW

**Comment by brickgriff** [comment]
What is the core concept for this PR?

File: src/app/app.component.html [hunk 1]
```diff
@@ -2,8 +2,10 @@
   <div class="spinner"></div>
 </div>
 <ion-app>
-  <ion-split-pane>
-    <ion-menu side="end">
+  <ion-split-pane contentId="main-content">
+    <div id="main-content">
+    <ion-menu side="end" contentId="my-content">
+      <div id="my-content"></div>
```
Comment (brickgriff) [inline src/app/app.component.html:8]: What is the purpose of this change?

File: src/app/pages/authentication/login/login.page.html [hunk 2]
```diff
@@ -3,7 +3,7 @@
 <ion-content>
   <form [formGroup]="signInForm" (ngSubmit)="submitForm()" novalidate>
 
-    <div padding class="span">
+    <div class="span ion-padding">
```
Comment (brickgriff) [inline src/app/pages/authentication/login/login.page.html:6]: So this is the new valid format?

File: src/app/pages/authentication/login/login.page.html [hunk 3]
```diff
@@ -21,7 +21,7 @@
         <ion-icon name="alert" color="danger"></ion-icon>Please enter your mobile number or email address.
       </span>
       <span class="error ion-padding" *ngIf="isSubmitted && errorControl.username.errors?.pattern || errorControl.username.errors?.minLength">
-        <ion-icon name="alert" color="danger"></ion-icon>Please provide your complete mobile number of email address.
+        <ion-icon name="alert" color="danger"></ion-icon>Please provide your complete mobile number or email address.
```
Comment (brickgriff) [inline src/app/pages/authentication/login/login.page.html:24]: Nice!

File: src/app/pages/card-import/card-import.page.ts [hunk 4]
```diff
@@ -51,11 +56,13 @@ export class CardImportPage implements OnInit {
            console.log("got card: ", card);
            this.card = card;
          });
-       } else {
+       } else if (this.id) {
```
Comment (brickgriff) [inline src/app/pages/card-import/card-import.page.ts]: This needs to be part of the above block. This code will never be called because it has the same condition as the previous one.

File: src/app/pages/card-import/card-import.page.ts [hunk 5]
```diff
@@ -129,7 +136,7 @@ export class CardImportPage implements OnInit {
       };
 
       this.cardService.addCard(this.card).then(() => {
-         this.router.navigate(["/profile"], navigationExtras);
+         this.router.navigate(["/home/home/profile"], navigationExtras);
```
Comment (brickgriff) [inline src/app/pages/card-import/card-import.page.ts]: Remember: the first argument here refers to the URL of the page, not the path to the file itself. There is already the line `{path: 'profile', loadChildren: './pages/home/profile/profile.module#ProfilePageModule'}` in `app-routing.module.ts` which defines all the global URLs. Either remove the global or use it. Finally, if this is the real URL for the nested Profile page, then it needs to be fixed. There shouldn't be duplicate "home" nested route path nodes. At most, the URL should say `"/home/profile"`.

File: src/app/pages/card-import/card-import.page.ts [hunk 6]
```diff
@@ -138,7 +145,7 @@ export class CardImportPage implements OnInit {
 
   deleteCard() {
      this.cardService.deleteCard(this.card.id).then(() => {
-       this.router.navigateByUrl('/profile');
+       this.router.navigateByUrl('/home/home/profile');
```
Comment (brickgriff) [inline src/app/pages/card-import/card-import.page.ts:152]: Again, this should be `"/home/profile"` at the most.

File: src/app/pages/card-import/card-import.page.ts [hunk 7]
```diff
@@ -170,13 +177,23 @@ export class CardImportPage implements OnInit {
 
   updateCard() {
       this.cardService.updateCard(this.card).then(() => {
-      this.router.navigateByUrl('/profile');
+      this.router.navigateByUrl('/home/home/profile');
```
Comment (brickgriff) [inline src/app/pages/card-import/card-import.page.ts:184]: Again, this should be `"/home/profile"` at the most.

File: src/app/pages/card-import/card-import.page.ts [hunk 8]
```diff
@@ -170,13 +177,23 @@ export class CardImportPage implements OnInit {
...
         this.showToast('Card updated');
       }, err => {
         this.showToast('There was a problem updating your card :(');
       });
   }
 
+  sendCard() {
```
Comment (brickgriff) [inline src/app/pages/card-import/card-import.page.ts:191]: Why is there a `sendCard()` feature in the Card Import page? How does the verb "send" match the intent of a page labeled "import"?

File: src/app/pages/card-import/card-import.page.html [hunk 9]
```diff
@@ -50,5 +51,10 @@
       </ion-button>
     </ion-col>
   </ion-row>
+    <ion-button expand="full" fill="solid" color="secondary" (click)="sendCard()">
```
Comment (brickgriff) [inline src/app/pages/card-import/card-import.page.html:54]: If general details and management has become the main focus of this page, then it should be split up and renamed. We should not have a page which shows existing card details and allows CRUD+Send actions named as "Card Import". Avoid overloading page responsibilities (Single Responsibility Principle). If we need a page for imports and a separate page for modifying and manipulating cards, then we should do that.

File: src/app/app-routing.module.ts [hunk 10]
```diff
@@ -37,11 +37,13 @@ const routes: Routes = [
   { path: 'edit', loadChildren: './pages/edit/edit.module#EditPageModule'},
 
   { path: 'card-import', loadChildren: './pages/card-import/card-import.module#CardImportPageModule' },
+  { path: 'card-roll', loadChildren: './pages/card-roll/card-roll/card-roll.module#CardRollPageModule' },
```
Comment (brickgriff) [inline src/app/app-routing.module.ts:40]: Didn't notice it the first time but this file is nested one level deeper than necessary.

File: src/app/pages/home/exchange/exchange.page.ts [hunk 11]
```diff
@@ -9,18 +17,169 @@ import { Router, NavigationExtras, ActivatedRoute } from '@angular/router';
...
+  private users: Observable<User[]>;
+  private cards: Observable<Card[]>;
 
-constructor(private route: ActivatedRoute, private router: Router) { }
-  addContact() {
-    let navigationExtras: NavigationExtras = {
-      state: {
-        parent: 'exchange'
+  id;
+  card: Card = {
```
Comment (brickgriff) [inline src/app/pages/home/exchange/exchange.page.ts:24]: Use the Card from CardService

File: src/app/pages/home/profile/profile.page.ts [hunk 12]
```diff
@@ -41,8 +42,110 @@ export class ProfilePage implements OnInit {
     targetHeight: 374
   }
 
+  slideOpts = {
```
Comment (brickgriff) [inline src/app/pages/home/profile/profile.page.ts:45]: There are examples of these configuration objects throughout the code. This code duplication needs to be moved to an external service. It takes up lots of space, is hard to read, and requires repeated updates in the future.

File: src/app/pages/home/profile/profile.page.ts [hunk 13]
```diff
@@ -62,18 +165,19 @@ export class ProfilePage implements OnInit {
...
 
   ngOnInit() {
-      console.log("User List page started (init)");
+      console.log("Profile page (init)");
 
       if (this.router.getCurrentNavigation().extras.state){
           this.card.frontImg = this.router.getCurrentNavigation().extras.state.cardDataFront;
           this.card.backImg = this.router.getCurrentNavigation().extras.state.cardDataBack;
-      };
+      } else if (this.isFrontCaptured = true){
```
Comment (brickgriff) [inline src/app/pages/home/profile/profile.page.ts]: This says "if the front image captured boolean is true, then load users and cards". What is the purpose of this conditional block?

File: src/app/pages/home/rolodex/rolodex.page.html [hunk 14]
```diff
@@ -1,35 +1,61 @@
-<ion-toolbar>
-   <ion-searchbar animated="false" placeholder="Search cards" [(ngModel)]="searchTerm"></ion-searchbar>
+<ion-toolbar class="header">
+  <ion-buttons slot="primary">
+    <ion-menu-button></ion-menu-button>
+  </ion-buttons>
+  <ion-searchbar animated="false" placeholder="Search cards" [(ngModel)]="searchTerm" *ngIf="!portraitCardView"></ion-searchbar>
 </ion-toolbar>
+<ion-searchbar animated="false" placeholder="Search cards" [(ngModel)]="searchTerm" *ngIf="portraitCardView"></ion-searchbar>
```
Comment (brickgriff) [inline src/app/pages/home/rolodex/rolodex.page.html:7]: Why the duplication?

File: src/app/pages/home/rolodex/rolodex.page.ts [hunk 15]
```diff
@@ -128,19 +133,67 @@ export class RolodexPage implements OnInit {
...
+    });
+    this.getGyroscopeData();
   }
 
   ngOnInit() {
     console.log("Rolodex Page (init)");
-    this.users = this.userService.listUsers();
   }
 
+  getScreenOrientation(){
```
Comment (brickgriff) [inline src/app/pages/home/rolodex/rolodex.page.ts:158]: Can we borrow these methods from the parent `home.page.ts`?

File: src/app/pages/menu/users/user-details/user-details.page.ts [hunk 16]
```diff
@@ -42,6 +42,10 @@ export class UserDetailsPage implements OnInit {
 
   }
 
+  addCard() {
```
Comment (brickgriff) [inline src/app/pages/menu/users/user-details/user-details.page.ts:45]: There is no need for an empty method, especially one designed to add a card in the user details page

**Review by brickgriff — CHANGES_REQUESTED** [review]
There are a number of critical changes and a few minor changes required for approval. Most of all, these sorts of omnibus branches will be rejected in the future. If more than three page transitions are required to demonstrate the change, consider splitting the change into multiple branches. 

For example, a login flow requires navigating to the login page, entering credentials, and navigating to the home page. Sign-up flow requires navigating to the registration page, then navigating to the user creation page, and finally passing through the success message to the home page.

This PR seems to have changes relating to gyroscope, screen orientation, and migration preparation/front-end styling for the card import page, user details page, all three home pages, and more. Pick one new feature or enhancement to merge at a time and resist immediately fixing unrelated bugs as these would be good issues for another branch.

This isn't to say the more branches the better. Keeping the purpose and scope of an issue, branch, and pull request focused and easily conveyed makes the planning, development, review, and coordination that much smoother.

File: src/app/app.component.html (hunk 1 above)
Comment (brandlxee11) [reply src/app/app.component.html:8]: A warning was being thrown that specified the prior format of the code was deprecated. Therefore, I changed it to match their recommended update. It was advised to eliminate warnings in preparation for migrating to Ionic 5.

File: src/app/app-routing.module.ts (hunk 10 above)
Comment (brandlxee11) [reply src/app/app-routing.module.ts:40]: It is. I think that can be changed lol

File: src/app/pages/card-import/card-import.page.ts (hunk 4 above)
Comment (brandlxee11) [reply src/app/pages/card-import/card-import.page.ts]: Great point!

File: src/app/pages/card-import/card-import.page.ts (hunk 5 above)
Comment (brandlxee11) [reply src/app/pages/card-import/card-import.page.ts]: Verdict: the URL for the nested Profile page needs to be fixed.

File: src/app/pages/card-import/card-import.page.ts (hunk 8 above)
Comment (brandlxee11) [reply src/app/pages/card-import/card-import.page.ts:191]: Because apparently I'm defying the Single Responsibility Principle and the card import page should be split up.

File: src/app/pages/card-import/card-import.page.ts (hunk 6 above)
Comment (brandlxee11) [reply src/app/pages/card-import/card-import.page.ts:152]: the URL for the nested Profile page needs to be fixed.

File: src/app/pages/card-import/card-import.page.ts (hunk 7 above)
Comment (brandlxee11) [reply src/app/pages/card-import/card-import.page.ts:184]: The URL for the nested Profile page needs to be fixed.

File: src/app/pages/home/profile/profile.page.ts (hunk 12 above)
Comment (brandlxee11) [reply src/app/pages/home/profile/profile.page.ts:45]: So we should definitely plot out when to tackle these refactoring points.

File: src/app/pages/home/profile/profile.page.ts (hunk 13 above)
Comment (brandlxee11) [reply src/app/pages/home/profile/profile.page.ts]: (sigh) To load users and cards when there is a card that is proven to have a front image captured? Honestly, I don't think it serves any true purpose.

File: src/app/pages/home/rolodex/rolodex.page.html (hunk 14 above)
Comment (brandlxee11) [reply src/app/pages/home/rolodex/rolodex.page.html:7]: Two different views based on the the physical orientation of the phone. The search bar moves to the toolbar in landscape to make cards fully visible on the screen.

File: src/app/pages/home/rolodex/rolodex.page.ts (hunk 15 above)
Comment (brandlxee11) [reply src/app/pages/home/rolodex/rolodex.page.ts:158]: I don't see why not. Via an external service?

File: src/app/pages/menu/users/user-details/user-details.page.ts (hunk 16 above)
Comment (brandlxee11) [reply src/app/pages/menu/users/user-details/user-details.page.ts:45]: Probably started down a path that I realized didn't need to be traveled.

File: src/app/app.component.html (hunk 1 above)
Comment (brickgriff) [reply src/app/app.component.html:8]: Okay but let's keep these changes in the proper scope. Reformatting/migration changes can be their own issue since it's a cross-cutting concern that affects all future issues/branches.

File: src/app/pages/card-import/card-import.page.ts (hunk 4 above)
Comment (bgriffiniv) [reply src/app/pages/card-import/card-import.page.ts:59]: Is this fixed?

File: src/app/pages/card-import/card-import.page.ts (hunk 5 above)
Comment (bgriffiniv) [reply src/app/pages/card-import/card-import.page.ts:139]: Is this fixed?

File: src/app/pages/card-import/card-import.page.ts (hunk 7 above)
Comment (bgriffiniv) [reply src/app/pages/card-import/card-import.page.ts:180]: Is this fixed?

File: src/app/pages/home/profile/profile.page.ts (hunk 12 above)
Comment (brickgriff) [reply src/app/pages/home/profile/profile.page.ts:45]: Yes this is probably an issue in its own right.

File: src/app/pages/home/profile/profile.page.ts (hunk 13 above)
Comment (brickgriff) [reply src/app/pages/home/profile/profile.page.ts]: But why are we loading all users and cards when a front image is taken? What's the connection?

File: src/app/pages/home/rolodex/rolodex.page.html (hunk 14 above)
Comment (brickgriff) [reply src/app/pages/home/rolodex/rolodex.page.html:7]: Is this really how it works? Why would you need two elements? Why wouldn't the one element shift in position? Are we hiding/showing each element based on orientation? That seems like overkill if they naturally fill the width of the containing element/screen. I'll need to see this in action.

File: src/app/pages/home/rolodex/rolodex.page.ts (hunk 15 above)
Comment (brickgriff) [reply src/app/pages/home/rolodex/rolodex.page.ts:158]: That would actually be better. Set up a screen service and gyroscope service and share their methods to any page that needs it with an import like any other service.

File: src/app/pages/card-import/card-import.page.html (hunk 9 above)
Comment (brickgriff) [reply src/app/pages/card-import/card-import.page.html:54]: This is just like the user-details page. It may be useful to have a send button included with the update and delete buttons for an existing card (if id exists) and a only create function (if id does not exist). It's okay to make a case for your ideas.

File: src/app/pages/home/profile/profile.page.ts (hunk 13 above)
Comment (brandlxee11) [reply src/app/pages/home/profile/profile.page.ts]: None. Was trying to eliminate the alert popping up when cards already exist.

File: src/app/pages/home/rolodex/rolodex.page.ts (hunk 15 above)
Comment (brandlxee11) [reply src/app/pages/home/rolodex/rolodex.page.ts:158]: That should def a part of the next refactoring/code review.

File: src/app/app-routing.module.ts (hunk 10 above)
Comment (brickgriff) [reply src/app/app-routing.module.ts:40]: This still needs to be changed.

File: src/app/app.component.html [hunk 17]
```diff
@@ -2,17 +2,29 @@
...
+      <div class="user-info">
+
+
         <ion-toolbar>
-          <ion-title>Menu</ion-title>
+          <ion-title slot="start">
+            <ion-avatar button (click)="capturePhoto()">
```
Comment (brickgriff) [inline src/app/app.component.html:14]: Is this new? Why are you adding stuff at this stage? What is the purpose of a photo capture button in the side menu (app component)? Please remove this until a later issue.

File: src/app/app.component.html [hunk 18]
```diff
@@ -2,17 +2,29 @@
...
+              <ion-img src="./assets/imgs/avatar.png"></ion-img>
+            </ion-avatar>
+            Brandon Lee
```
Comment (brickgriff) [inline src/app/app.component.html:17]: Do not hard-code names like this unless prototyping a feature. If you are prototyping a feature, please create a branch specifically for it.

File: src/app/app.component.scss [hunk 19]
```diff
@@ -39,3 +38,9 @@
     -webkit-transform: perspective(120px) rotateX(-180deg) rotateY(-179.9deg);
   }
 }
+
+.user-info {
```
Comment (brickgriff) [inline src/app/app.component.scss:42]: This seems like yet another late-stage feature smuggling. Please remove this and save for a new issue.

File: src/app/app.component.ts [hunk 20]
```diff
@@ -100,4 +108,36 @@ export class AppComponent {
   goToHomePage() {
     this.router.navigateByUrl('/home');
   }
+
+  async takePhoto() {
```
Comment (brickgriff) [inline src/app/app.component.ts:112]: Where is this used?

File: src/app/app.component.ts [hunk 21]
```diff
@@ -100,4 +108,36 @@ export class AppComponent {
...
+         // this.camera.DestinationType.DATA_URL gives base64 URI
+
+         let base64Image = 'data:image/jpeg;base64,' + imageData;
+         this.profilePic = base64Image;
+       }, (err) => {
+         console.log("Unable to obtain picture: " + err);
+       });
+  }
+
+  async capturePhoto() {
```
Comment (brickgriff) [inline src/app/app.component.ts:124]: Please remove this and any other new changes and put them in new issues.

File: src/app/app.component.ts [hunk 22]
```diff
@@ -54,26 +56,32 @@ export class AppComponent {
...
-    },
-    */
     {
       title: 'User List',
       url: '/users',
       icon: 'aperture'
     }
   ];
 
+  cameraOptions: CameraOptions = {
```
Comment (brickgriff) [inline src/app/app.component.ts:66]: This is the kind of thing that should go in its own issue/branch. Why are camera options important for the side menu (app component)?

File: src/app/app.module.ts [hunk 23]
```diff
@@ -1,21 +1,25 @@
...
 
 import { IonicModule, IonicRouteStrategy } from '@ionic/angular';
 import { SplashScreen } from '@ionic-native/splash-screen/ngx';
 import { StatusBar } from '@ionic-native/status-bar/ngx';
-import { Camera } from '@ionic-native/camera/ngx';
 import { HttpClientModule } from '@angular/common/http';
 
 import { AppComponent } from './app.component';
 import { AppRoutingModule } from './app-routing.module';
+import { CardRollPageModule } from 'src/app/pages/card-roll/card-roll/card-roll.module';
```
Comment (brickgriff) [inline src/app/app.module.ts:12]: This is still deeply nested.

File: src/app/pages/card-import/card-import.page.ts [hunk 24]
```diff
@@ -19,6 +21,9 @@ export class CardImportPage implements OnInit {
     backImg: ''
```
Comment (brickgriff) [inline src/app/pages/card-import/card-import.page.ts:21]: For simplicity, please use the Card interface provided by the Card Service.

File: src/app/pages/card-import/card-import.page.ts [hunk 25]
```diff
@@ -37,13 +42,18 @@ export class CardImportPage implements OnInit {
...
+              private toastCtrl: ToastController,
+              public alertController: AlertController,
+              private cardService: CardService) {
 
   }
 
    ngOnInit() {
        this.id = this.route.snapshot.paramMap.get('id');
-       console.log(this.id)
+       console.log(this.id);
```
Comment (brickgriff) [inline src/app/pages/card-import/card-import.page.ts:56]: Copy changes from or rebase onto dev to get the changes here. When `this.id` exists, then the page should pull the corresponding card data. Currently, `this.card` does not exist.

File: src/app/pages/card-import/card-import.page.ts (hunk 6 above)
Comment (brickgriff) [reply src/app/pages/card-import/card-import.page.ts:152]: It is likely that this page does not know about `/home/profile`. Simply using `/profile` which is available globally in `app-routing.module.ts` works just as well. Focus on restructuring the `/home` sub-route later as a new issue.

File: src/app/pages/card-import/card-import.page.ts (hunk 7 above)
Comment (brickgriff) [reply src/app/pages/card-import/card-import.page.ts:184]: It is likely that this page does not know about /home/profile. Simply using /profile which is available globally in app-routing.module.ts works just as well. Focus on restructuring the /home sub-route later as a new issue.

File: src/app/pages/card-import/card-import.page.ts [hunk 26]
```diff
@@ -129,7 +140,7 @@ export class CardImportPage implements OnInit {
       };
 
       this.cardService.addCard(this.card).then(() => {
-         this.router.navigate(["/profile"], navigationExtras);
+         this.router.navigate(["/home/profile"], navigationExtras);
```
Comment (brickgriff) [inline src/app/pages/card-import/card-import.page.ts:143]: It is likely that this page does not know about /home/profile. Simply using /profile which is available globally in app-routing.module.ts works just as well. Focus on restructuring the /home sub-route later as a new issue.

File: src/app/pages/card-import/card-import.page.ts (hunk 8 above)
Comment (brickgriff) [reply src/app/pages/card-import/card-import.page.ts:191]: That's less important than determining whether this Pull Request is about card exchange, card import, screen orientation, or formatting. This does not match what we've discussed. We should be sending `this.id`, at most, to determine which card should add the transmitted new holder uid to its holder list (this will happen in the exchange page). This can be removed until that feature is ready for work.

File: src/app/pages/card-import/card-import.page.ts [hunk 27]
```diff
@@ -170,13 +181,23 @@ export class CardImportPage implements OnInit {
...
   }
 
+  sendCard() {
+     let navigationExtras: NavigationExtras = {
+        state: {
+            cardDataFront: this.card.frontImg,
+            cardDataBack: this.card.backImg
+        }
+     };
+     this.router.navigate(["/home/home/exchange"], navigationExtras);
```
Comment (brickgriff) [inline src/app/pages/card-import/card-import.page.ts:198]: Also the global `/exchange` path exists in `app-routing.module.ts`.

File: src/app/pages/home/exchange/exchange.page.ts (hunk 11 above)
Comment (brickgriff) [reply src/app/pages/home/exchange/exchange.page.ts:24]: Don't just give a thumbs up! Remove the rest of this type alias!!!

File: src/app/pages/home/rolodex/rolodex.page.html (hunk 14 above)
Comment (brickgriff) [reply src/app/pages/home/rolodex/rolodex.page.html:7]: Assuming this doesn't break compilation, I'd say this is resolved until we can figure out whether this is really valid during integration testing (dev).

**Review by brickgriff — CHANGES_REQUESTED** [review]
This PR is far too large to actually verify as a unit of work. I think we can merge some of these but it will require heavy integration testing in `dev`. There are a few changes that still need to be made for approval.  After that, we can set up a roadmap for breaking this down and making the needed fixes in order. This is "fine" only because we don't have a `prod` yet and our `dev` is kind of pulling double or triple duty. In the future, we want to be testing units in feature branches and testing integrations in `dev`. 

Stop adding to this branch, make the changes I've requested, let's get this merged, find the weaknesses in `dev`, write up a backlog of bugs, and tackle them in their own branches. Submitting PRs with this many changed files gives us more work to do than when we started.

File: src/app/pages/menu/users/user-details/user-details.page.ts (hunk 16 above)
Comment (brickgriff) [reply src/app/pages/menu/users/user-details/user-details.page.ts:45]: Then please remove it!

**Comment by brickgriff** [comment]
Since resolving the current conflicts, while relatively simple, would require blending this branch with `dev` (rather than selecting one over the other), I am closing this PR. Please stash these changes in another branch, recreate this branch, and submit a concise and focused new PR.

File: src/app/pages/card-import/card-import.page.ts (hunk 6 above)
Comment (brandlxee11) [reply src/app/pages/card-import/card-import.page.ts:152]: I figure the most taxing part of this process at the moment is building a cohesive foundation of the MVP so we can then collectively navigate the app ourselves to address blind spots and gaps, fix bugs, and give the design structure and interface the necessary face lifts.

File: src/app/pages/card-import/card-import.page.ts (hunk 7 above)
Comment (brandlxee11) [reply src/app/pages/card-import/card-import.page.ts:184]: I figure the most taxing part of this process at the moment is building a cohesive foundation of the MVP so we can then collectively navigate the app ourselves to address blind spots and gaps, fix bugs, and give the design structure and interface the necessary face lifts

File: src/app/app.component.html (hunk 17 above)
Comment (brandlxee11) [reply src/app/app.component.html:14]: Yes it is. I was playing around with a storyboard scenario for the exchange process where when two people roll each other their "app picture" would pop up to notify each other that their phone's sense each other through the NFC.

File: src/app/app.component.html (hunk 18 above)
Comment (brandlxee11) [reply src/app/app.component.html:17]: This was prototyping a feature. See above explanation.

File: src/app/pages/card-import/card-import.page.ts (hunk 8 above)
Comment (brandlxee11) [reply src/app/pages/card-import/card-import.page.ts:191]: It's quite the mental workout bouncing back and forth between creative storytelling and the codification of it. Wires get crossed a lot. But hey, that's what the team's for right?

File: src/app/app-routing.module.ts (hunk 10 above)
Comment (brandlxee11) [reply src/app/app-routing.module.ts:40]: We'll need to revamp file nestings. That can be its own issue too.

File: src/app/pages/home/exchange/exchange.page.ts (hunk 11 above)
Comment (brandlxee11) [reply src/app/pages/home/exchange/exchange.page.ts:24]: Lol bruh! I think that's best done during code review. Part of me doesn't trust changing code that I've conceived needs to be there based on the work I've done. I know you made the change in the service, but I still feel unsure. For this matter, I think we should always have a demo when we complete an issue. It ensures we each understand what features and functionality are being added.

File: src/app/app.component.scss (hunk 19 above)
Comment (brandlxee11) [reply src/app/app.component.scss:42]: Yeah, that was part of my prototyping. It can go.

File: src/app/app.component.ts (hunk 20 above)
Comment (brandlxee11) [reply src/app/app.component.ts:112]: This was a method created for prototyping the user profile picture capture in the menu. 
Should've been placed in another issue.

File: src/app/app.component.ts (hunk 21 above)
Comment (brandlxee11) [reply src/app/app.component.ts:124]: Sounds good.

File: src/app/app.component.ts (hunk 22 above)
Comment (brandlxee11) [reply src/app/app.component.ts:66]: I was prototyping again. But I'll admit at this stage I was getting lost in the storytelling, so I started experimenting to see what different changes looked like. It was at this point that I stepped away from developing for a couple days. I should've reverted my changes, but I was fatigued at the time. Yet another lesson learned.

File: src/app/app.module.ts (hunk 23 above)
Comment (brandlxee11) [reply src/app/app.module.ts:12]: Yes, it can be unnested. This can go in its own issue.

File: src/app/pages/card-import/card-import.page.ts (hunk 24 above)
Comment (brandlxee11) [reply src/app/pages/card-import/card-import.page.ts:21]: Yep, these are the changes that need to be made as a result of your changes in the card service.

File: src/app/pages/card-import/card-import.page.ts (hunk 25 above)
Comment (brandlxee11) [reply src/app/pages/card-import/card-import.page.ts:56]: Yes, I saw you made that change in one of the branches previously merged branches. Admittedly, I didn't understand the purpose behind the change. I suggest a follow-up conversation where you specify these details so we all understand clearly.

File: src/app/pages/card-import/card-import.page.ts (hunk 27 above)
Comment (brandlxee11) [reply src/app/pages/card-import/card-import.page.ts:198]: A "nesting issue" is definitely in order.

File: src/app/app.component.html (hunk 17 above)
Comment (brickgriff) [reply src/app/app.component.html:14]: Yeah don't add to a pending pull request under review. This should be its own issue, branch, and PR.

Comment (brandlxee11) [reply src/app/app.component.html:14]: No, no this was before the pull request I assure you.
